src/controllers/admin/user.controller.ts

```diff
@@ -0,0 +1,19 @@
+import type { Request, Response } from 'express'
+import { getAllUsers } from '../../services/admin.service'
+
+export const getAllUsersHandler = async (
+  req: Request,
+  res: Response
+): Promise<any> => {
+  try {
+    const users = await getAllUsers(req)
```

Comment:
```suggestion
    const users = await getAllUsers()
```

You need to check the user type here. user type should be ADMIN otherwise throw 403 unauthorized